picard/tagger.py

```diff
@@ -341,6 +347,7 @@ def exit(self):
         self._acoustid.done()
         self.thread_pool.waitForDone()
         self.save_thread_pool.waitForDone()
+        self.priority_thread_pool.waitForDone()
```

Comment:
Since we have more thread pools now, i'd rather use a dict to store them:

```python
self.threads = ('main', 'priority', 'save')
self.thread_pools = dict()
for thread in self.threads:
    self.thread_pools[thread] = QtCore.QThreadPool(self)
...

for thread in self.threads:
    self.thread_pools[thread].waitForDone()
```

And perhaps change `run_task()` in https://github.com/metabrainz/picard/blob/master/picard/util/thread.py to use the thread name instead.

Of course we could use a class instead of a dict to make it cleaner.

Reply:
I am actually not too sure I like the dict approach better than the three attributes we have right now. Also we should also probably take care that this does not run out of control and we end up managing a pool of pools :)